### PR TITLE
feat: make frontend aware that OIDC can be configured through env

### PR DESCRIPTION
**Pull request by chriswk**

### CODE REVIEW

File: frontend/src/component/admin/auth/OidcAuth/OidcAuth.tsx [hunk 1]
```diff
@@ -21,6 +21,7 @@ import { formatUnknownError } from 'utils/formatUnknownError';
 import { removeEmptyStringFields } from 'utils/removeEmptyStringFields';
 import { SsoGroupSettings } from '../SsoGroupSettings';
 import type { IRole } from 'interfaces/role';
+import { ConditionallyRender } from '../../../common/ConditionallyRender/ConditionallyRender';
```
Comment (nunogois) [inline frontend/src/component/admin/auth/OidcAuth/OidcAuth.tsx:24]: We usually prefer this:
```suggestion
import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
```

File: frontend/src/component/admin/auth/OidcAuth/OidcAuth.tsx [hunk 2]
```diff
@@ -45,10 +46,10 @@ type State = typeof initialState & {
 export const OidcAuth = () => {
     const { setToastData, setToastApiError } = useToast();
     const { uiConfig } = useUiConfig();
+    const { oidcLocked } = uiConfig;
     const [data, setData] = useState<State>(initialState);
     const { config } = useAuthSettings('oidc');
     const { updateSettings, errors, loading } = useAuthSettingsApi('oidc');
-
     useEffect(() => {
```
Comment (nunogois) [inline frontend/src/component/admin/auth/OidcAuth/OidcAuth.tsx:53]: Poor empty line got yeeted, but I think it helps with readability / separation of logic blocks 😄 
```suggestion
    
    useEffect(() => {
```

File: frontend/src/interfaces/uiConfig.ts [hunk 3]
```diff
@@ -30,6 +30,7 @@ export interface IUiConfig {
     strategySegmentsLimit?: number;
     frontendApiOrigins?: string[];
     resourceLimits: ResourceLimitsSchema;
+    oidcLocked?: boolean;
```
Comment (nunogois) [inline frontend/src/interfaces/uiConfig.ts:33]: Now that I think about it, maybe we should be a bit more specific about this name. It doesn't really tell us why OIDC is locked. How about something like `oidcConfiguredThroughEnv`?

File: frontend/src/component/admin/auth/OidcAuth/OidcAuth.tsx [hunk 4]
```diff
@@ -107,6 +108,24 @@ export const OidcAuth = () => {
         <>
             <Grid container sx={{ mb: 3 }}>
                 <Grid item md={12}>
+                    <ConditionallyRender
+                        condition={oidcLocked === true}
```
Comment (nunogois) [inline frontend/src/component/admin/auth/OidcAuth/OidcAuth.tsx:112]: No strong feelings either way, just an alternative.
```suggestion
                        condition={Boolean(oidcLocked)}
```

**Review by nunogois — APPROVED** [review]
LGTM!

I think maybe we should rethink the variable name to be more explicit / descriptive of what's really going on. 

Other than that, I added some small comments that are not very significant, so feel free to ignore them.

File: frontend/src/component/admin/auth/OidcAuth/OidcAuth.tsx (hunk 1 above)
Comment (chriswk) [reply frontend/src/component/admin/auth/OidcAuth/OidcAuth.tsx:24]: Yeah, auto imports. will update

File: frontend/src/interfaces/uiConfig.ts (hunk 3 above)
Comment (chriswk) [reply frontend/src/interfaces/uiConfig.ts]: Took your suggestion. It made sense to me. Updated.

**Pull request merged**